### PR TITLE
feat: Additional check for existence of Composer autoloader

### PR DESCRIPTION
**Pull request by kampalex**
# Fixes #
The `sendgrid-php.php` expects to have the Composer installed `vendor/autoload.php`.
This PR will verify its existence, and creates an error message if Composer autoloader (or configuration) isn't installed or failing.
(Execution of script itself won't be terminated.)

The use of above filenames is mentioned in the updated docs.
As I was working in this file (`CONTRIBUTING.md`), I've fixed a bookmark link and updated PHP version prerequisites.

### Checklist
- [X] I acknowledge that all my contributions will be made under the project's license
- [X] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [X] I have read the [Contribution Guidelines](CONTRIBUTING.md) and my PR follows them
- [X] I have titled the PR appropriately
- [X] I have updated my branch with the master branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation about the functionality in the appropriate .md file
- [X] I have added inline documentation to the code I modified

### CODE REVIEW

File: sendgrid-php.php [hunk 1]
```diff
@@ -1,3 +1,28 @@
...
+
+// Can't locate SendGrid\Mail\Mail class?
+if (!class_exists(Mail::class)) {
+    // Suggest to load Composer autoloader of project
+    if (!$composerAutoloadFileExists) {
+        //  Can't load the Composer autoloader in this project folder
+        error_log("Composer autoloader not found. Execute 'composer install' in the project folder first.");
+    } else {
+        // Load Composer autoloader
+        require_once $composerAutoloadFile;
```
Comment (thinkingserious) [inline sendgrid-php.php:20]: Why the change from `require` to `require_once`? Would this break anything?

File: sendgrid-php.php [hunk 2]
```diff
@@ -1,3 +1,28 @@
 <?php
-require __DIR__ . '/vendor/autoload.php';
-?>
\ No newline at end of file
+/**
+ * This file is used to load the Composer autoloader if required.
```
Comment (thinkingserious) [inline sendgrid-php.php:3]: 👍

File: sendgrid-php.php [hunk 3]
```diff
@@ -1,3 +1,28 @@
 <?php
-require __DIR__ . '/vendor/autoload.php';
-?>
\ No newline at end of file
+/**
```
Comment (thinkingserious) [inline sendgrid-php.php:2]: Please add tests for this file.

File: sendgrid-php.php [hunk 4]
```diff
@@ -1,3 +1,28 @@
...
+        // Load Composer autoloader
+        require_once $composerAutoloadFile;
+
+        // If desired class still not existing
+        if (!class_exists(Mail::class)) {
+            // Suggest to review the Composer autoloader settings
+            error_log("Error finding SendGrid classes. Please review your autoloading configuration.");
+        }
+    }
+}
```
Comment (thinkingserious) [inline sendgrid-php.php:28]: I saw this PR https://github.com/sendgrid/sendgrid-php/pull/984 before this one. Now, after reviewing this one, I think we should merge this first and then revisit the wording in #984.

File: sendgrid-php.php (hunk 1 above)
Comment (kampalex) [reply sendgrid-php.php:20]: It's personal preference, but mostly used when loading files which contains functions/classes.
Not breaking because `$composerAutoloadFile` will be loaded anyway, but just once.
No big thing to change back to `require`.

File: sendgrid-php.php (hunk 3 above)
Comment (kampalex) [reply sendgrid-php.php:2]: Testing functionality of it is difficult. This is because Composer is already loaded, and class `SendGrid\Mail\Mail` exists at that time.
Adding an `assertFileExists` in `FilesExistTest` is the only scenario which gets close to a - so called - 'test'.

File: sendgrid-php.php (hunk 4 above)
Comment (kampalex) [reply sendgrid-php.php:28]: Exactly. This PR gives the base to optimize #984, but without it this PR is useless to implement (chicken or the egg).

**Comment by kampalex** [comment]
Important to mention:

In the `scripts/package.sh` file (which creates a release package), contents of `sendgrid-php.php` are overwritten by `<?php\nrequire __DIR__ . '/vendor/autoload.php';\n?>`.
This means that the additional functionality (of this PR) is **only available when pulling/forking the repository**.

Since the release package contains the Composer autoloader, there is no need to perform additional checks there.
